### PR TITLE
🌱 Add controller-name as kv pair to logger instead of as name

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

Addresses #1794. More context in the initial post of #1794

### CODE REVIEW

**Review by alvaroaleman — APPROVED** [review]
I like the idea but it is a breaking change as existing log queries will stop working. Are we comfortable with that?
/hold

File: pkg/controller/controller.go [hunk 1]
```diff
@@ -135,7 +135,7 @@ func NewUnmanaged(name string, mgr manager.Manager, options Options) (Controller
 		CacheSyncTimeout:        options.CacheSyncTimeout,
 		SetFields:               mgr.SetFields,
 		Name:                    name,
-		Log:                     options.Log.WithName("controller").WithName(name),
+		Log:                     options.Log.WithValues("controller", name),
```
Comment (sbueringer) [inline pkg/controller/controller.go:138]: @alvaroaleman 
Maybe we can set the name and k/v?

I just saw this user story in the kubernetes contextual logging KEP:
> When kube-scheduler starts processing a pod, it creates a new logger with logger.WithValue("pod", klogr.KObj(pod)) and passes that around. While iterating over plugins in certain operations, another logger gets created with logger.WithName(<operation>).WithName(<plugin name>) and then is used when invoking that plugin. This adds a prefix to each log message which represents the call path to the specific log message, like for example NominatedPods/Filter/VolumeBinding.
> https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/3077-contextual-logging#story-1

To me it seemed strange to have a msg prefix with the controller. But looking at that user story it kind of makes sense.

Comment (alvaroaleman) [reply pkg/controller/controller.go:138]: Yeah, that makes sense and seems a good idea

**Review by alvaroaleman — APPROVED** [review]
thanks!
/hold cancel

**Pull request merged**